### PR TITLE
remove duplicate config_loader creation when -t is enabled

### PR DESCRIPTION
**Pull request by jsvd**
fixes https://github.com/elastic/logstash/issues/5764

### CODE REVIEW

File: logstash-core/spec/logstash/agent_spec.rb [hunk 1]
```diff
@@ -313,6 +313,11 @@
         expect(subject.run(cli_args)).to eq(1)
       end
     end
+
+    it "requests the config loader to format_config" do
+      expect(subject.config_loader).to receive(:format_config)
+      subject.run(cli_args)
+    end
```
Comment (ph) [inline logstash-core/spec/logstash/agent_spec.rb]: Should we actually test that the config was correctly sent to the logger?

**Comment by ph** [comment]
I've tested this PR and I got what I believe a weird result.

if you do `bin/logstash --debug --debug-config -t  -e "input { generator {}}"` you wont get any output of the configuration.

If you do `bin/logstash --debug --debug-config -t -f hello.conf -e "input { tcp { port => 9000}}"` You will see only the content of the *hello.conf*, but the pipeline will also setup the TCP input..

**Comment by jsvd** [comment]
Well spotted. Interestingly, the first scenario has always been true since --debug-config was introduced, the piece of config coming from -e isn't included in debug. I'll check what's needed for it, but IMO can be done in another PR.

As for the second thing..it doesn't actually open the tcp port, just does the plugin init (not register). This has happened in older logstash versions.

**Comment by ph** [comment]
> Well spotted. Interestingly, the first scenario has always been true since --debug-config was introduced, the piece of config coming from -e isn't included in debug. I'll check what's needed for it, but IMO can be done in another PR.

Agree for another PR.

> As for the second thing..it doesn't actually open the tcp port, just does the plugin init (not register). This has happened in older logstash versions.

oh that is effectively confusing then.

**Comment by ph** [comment]
LGTM, this PR fix the mentioned issue.

**Pull request closed without merging**